### ISSUE TITLE
The node loss function

### ISSUE THREAD

**Issue by Edwarddd** [snudatalab/TAG#2]
I noticed that the node loss function is incorrect. 
```
select_num = int(step ** lamb * node_num)
select_idx = torch.tensor(np.arange(select_num)).to(self.device)
loss = torch.index_select(scores, -1, select_idx)
loss = - torch.log(loss).sum()
```

It leads to a continuous increase in the loss. 

I'm not sure if this is because of a typo, a `-torch.log(loss).mean()` would help.
